### PR TITLE
Upgrade solc version to 0.8.17

### PR DESCRIPTION
**Pull request by JonneyTheSheep**
### Description

* format code
* upgrade solidity compile to ^v0.8.0
* Replace  License intensifier to Apache-2.0 and MIT
* interface files add prefix with `I` aka `Interface`( e.g: `NFTFactory` to `INFTFactory`)
* explicit type conversion, e.g. uint256(self) to uint256(uint160(self`address`))
* uint change to uint256, because "uint and int are aliases for uint256 and int256, respectively." 
* More semantic e.g. requireGovernor to isGovernor
* Replace  RequireXXX function to onlyXXX modifier, (e.g., requireActive/requireMaster)

### CODE REVIEW

File: contracts/UpgradeableMaster.sol [hunk 1]
```diff
@@ -63,7 +60,7 @@ contract UpgradeableMaster {
   // Upgrade functional
   /// @notice Shortest Notice period before activation preparation status of upgrade mode
   ///         Notice period can be set by secure council
-  function getNoticePeriod() external view returns (uint256) {
+  function getNoticePeriod() external pure returns (uint256) {
```
Comment (rzrrbk) [inline contracts/UpgradeableMaster.sol:63]: Should we keep this function as `view`?  as it reads the state variable

File: contracts/ZkBNB.sol [hunk 2]
```diff
@@ -113,8 +103,8 @@ contract ZkBNB is Events, Storage, Config, ReentrancyGuardUpgradeable, IERC721Re
     stateRoot = _genesisStateRoot;
     storedBlockHashes[0] = hashStoredBlockInfo(zeroStoredBlockInfo);
   }
-
-  /// @notice ZkBNB contract upgrade. Can be external because Proxy contract intercepts illegal calls of this function.
+
```
Comment (rzrrbk) [inline contracts/ZkBNB.sol:106]: trailing whitespace

File: contracts/ZkBNBVerifier.sol [hunk 3]
```diff
@@ -1,6 +1,6 @@
 // SPDX-License-Identifier: AML
```
Comment (rzrrbk) [inline contracts/ZkBNBVerifier.sol]: hmm, couldn't find any information about this license, do we need to change it?

File: contracts/interfaces/Events.sol [hunk 4]
```diff
@@ -1,8 +1,6 @@
-// SPDX-License-Identifier: MIT OR Apache-2.0
+// SPDX-License-Identifier: MIT
```
Comment (rzrrbk) [inline contracts/interfaces/Events.sol:1]: MIT -> Apache-2.0 ?

Comment (rzrrbk) [reply contracts/interfaces/Events.sol:1]: Noted some files are under MIT while some are under Apache-2.0. Wondering what's our standard for choosing a license 💭

File: contracts/ZkBNBVerifier.sol (hunk 3 above)
Comment (rzrrbk) [reply contracts/ZkBNBVerifier.sol]: Found it. It's a variant of MIT. I think we can keep it. 

https://fedoraproject.org/wiki/Licensing/Apple_MIT_License

File: contracts/UpgradeableMaster.sol (hunk 1 above)
Comment (JonneyTheSheep) [reply contracts/UpgradeableMaster.sol:63]: https://docs.soliditylang.org/en/v0.8.17/070-breaking-changes.html#changes-to-the-syntax
public state variables are considered view and even pure if they are constants.

File: contracts/ZkBNBVerifier.sol (hunk 3 above)
Comment (JonneyTheSheep) [reply contracts/ZkBNBVerifier.sol]: Now, two files has AML liscense, Pairing.sol and ZkBNBVerifier.sol file
Pairing.sol come from [Christian Reitwiessner](https://github.com/iden3/snarkjs/blob/master/templates/verifier_groth16.sol.ejs), and use GPL3.0, so we should change to GPL 3.0 ,and add license comments.
As for the ZkBNBVerifier.sol , I can't find anymore, we can change to Apache 2.0

File: contracts/interfaces/Events.sol (hunk 4 above)
Comment (JonneyTheSheep) [reply contracts/interfaces/Events.sol:1]: `Interface` files could be used in other DApp, even not from zkBNB team
MIT license is more permissive

File: contracts/ZkBNBVerifier.sol (hunk 3 above)
Comment (rzrrbk) [reply contracts/ZkBNBVerifier.sol]: maybe the initial version of ZkBNBVerifier.sol is from https://github.com/HarryR/ethsnarks/blob/master/contracts/Verifier.sol, which is referenced by lots of zk projects and not restricted(?) by liscense.

**Review by qingyang-id — APPROVED** [review]

**Pull request merged**